internal/controllers/lvmcluster/resource/lvm_volumegroupnodestatus.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright © 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"context"
+	"fmt"
+
+	lvmv1alpha1 "github.com/openshift/lvm-operator/api/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	lvmVGNodeStatusName = "lvmvgnodestatus"
+)
+
+func LVMVGNodeStatus() Manager {
+	return lvmVGNodeStatus{}
+}
+
+type lvmVGNodeStatus struct{}
+
+// lvmVGNodeStatus unit satisfies resourceManager interface
+var _ Manager = lvmVGNodeStatus{}
+
+func (l lvmVGNodeStatus) GetName() string {
+	return lvmVGNodeStatusName
+}
+
+// EnsureCreated is a noop. This will be created by vg-manager.
+func (l lvmVGNodeStatus) EnsureCreated(r Reconciler, ctx context.Context, cluster *lvmv1alpha1.LVMCluster) error {
+	return nil
+}
+
+func (l lvmVGNodeStatus) EnsureDeleted(r Reconciler, ctx context.Context, _ *lvmv1alpha1.LVMCluster) error {
+	logger := log.FromContext(ctx).WithValues("resourceManager", l.GetName())
+	vgNodeStatusList := &lvmv1alpha1.LVMVolumeGroupNodeStatusList{}
+	if err := r.List(ctx, vgNodeStatusList, client.InNamespace(r.GetNamespace())); err != nil {
+		return fmt.Errorf("failed to list LVMVolumeGroupNodeStatus: %w", err)
+	}
+
+	for _, nodeItem := range vgNodeStatusList.Items {
+		if !nodeItem.GetDeletionTimestamp().IsZero() {
+			return fmt.Errorf("the LVMVolumeGroupNodeStatus %s is still present, waiting for deletion", nodeItem.GetName())
+		}
+		if err := r.Delete(ctx, &nodeItem); err != nil {
```

Comment:
I think it would be wise to delete it only if it does not contain any volume groups anymore since that is managed by vgmanager. WDYT?

Reply:
This comes after volume group removal (https://github.com/openshift/lvm-operator/pull/617/files#diff-3ddf36bfe9ab457a2b494fb56efbf5f1f70472c7e7f62d29779545f0a75ecff1R423) so the reconciler won't start removal of lvmvolumegroupnodestatus unless all the volume groups are gone. We could still add one more check here to double-check, but this will copy [this logic](https://github.com/openshift/lvm-operator/blob/main/internal/controllers/lvmcluster/resource/lvm_volumegroup.go#L93) so I don't know if it's worth it. Wdyt?

Reply:
I believe the resource deletion should be indempotent when doing a concurrent delete. I know we currently delete serialized but when we eventually switch to concurrent removal of all resources then we would need the check.